src/encoder.rs

```diff
@@ -2213,7 +2211,8 @@ fn encode_partition_bottomup<T: Pixel, W: Writer>(
   // Always split if the current partition is too large, i.e. right or bottom tile border
   let must_split = (tile_bo.0.x + bsw as usize > ts.mi_width
     || tile_bo.0.y + bsh as usize > ts.mi_height
-    || bsize > BlockSize::BLOCK_64X64)
+    || bsize > BlockSize::BLOCK_64X64
+    || bsize > fi.partition_range.max())
```

Comment:
I'm thinking 2214 can be removed. 2214 was the old version of 2215, because the max block size was implicitly 64x64 for all block sizes.